### ISSUE TITLE
[WCAG2.1]A11y_Java Tooling for IntelliJ/Eclipse_Spark Submission_Screen Reader: Narrator/NVDA is reading the "Browse" button as "Open system file dialog to select artifact" button

### ISSUE THREAD

**Issue by soumyatippapuram** [microsoft/azure-tools-for-java#4271]
[Check out Accessibility Insights! ](https://accessibilityinsights.io/)- Identify accessibility bugs before check-in and make bug fixing faster and easier

GitHubTags: #A11yTCS; #A11yWCAG2.1; #DesktopApp; #Win10; #A11y_JavaToolingforIntelliJ/Eclipse-Benchmark; #MAS2.5.3; #ScreenReader; #NVDA; #Narrator; #JavatoolingforIntelliJ/Eclipse; #C+E-H; #LabelInName; #Benchmark; #A11ySev1; #A11y_JavaToolingforIntelliJ/Eclipse-WCAG2.1

**Environment Details:**
Eclipse IDE for Enterprise Java Developers.
Version: 2020-03 (4.15.0)
Build id: 20200313-1211

**Solution Installed:** The Azure Toolkit for Eclipse 3.220, Scala IDE 4.7.x

**OS Details:**
OS: Windows (Version 1903, Build: 18362.175)

**Screen Reader:** Narrator, NVDA

**MAS Violated:** MAS 2.5.3

**Repro Steps:**
1.Start Screen Reader. Open Java EE - Eclipse application
2.Install "Azure Toolkit for Eclipse" and "Scala Plug-ins" from Eclipse Marketplace
3.Tab till "File > New > Project" in the navigation bar and press enter
4.Tab till "HDInsight Project > Spark on HDInsight (Scala)" and select it.
5.Tab till "Next" button and press enter 
6.Enter all the required details, tab till "Next" button and press enter
7.Tab till "Module Dependencies" tab and press enter
8.Tab till "Finish" button and press enter
9.In Project Explorer, expand the created Scala project right-click src, New-->Other and press enter
10.Tab till Scala Wizards, expand and select 'Scala Object' and press next
11.Open the created file and write any code
12.In Project Explorer view, open contextual menu and select "Submit Spark Application to HDInsight"
13.Tab till "Browse" button beside "Artifact from hard disk" radio button
14.Check if the screen reader is reading the "Browse" button properly

**Actual Result:**
Narrator/NVDA is reading the "Browse" button as "Open system file dialog to select artifact" button

**Expected Result:**
Narrator/NVDA should read the word "Browse" along with "Open system file dialog to select artifact" when the focus is on "Browser" button

**User impact:**
Speech-input users will find it difficult to access the "Browse" button 

**MAS Reference:**
[MAS 2.5.3](https://microsoft.sharepoint.com/:w:/r/sites/accessibility/_layouts/15/Doc.aspx?sourcedoc=%7B2035FD29-3ADF-4309-A770-8D572CA3B9B7%7D&file=MAS%202.5.3%20%E2%80%93%20Label%20in%20Name.docx&action=default&mobileredirect=true)

![Bug_4271_Java Tooling_Spark Submission_Label in Name_NVDA](https://user-images.githubusercontent.com/41108985/78909037-8d2b4280-7aa0-11ea-870a-04e2f76131e0.png)

**Comment by jingyanjingyan** [comment]
+ Importance of experience: Minimal. Speech-input users will find it difficult to access the "Browse" button.
+ Quality of experience: Manageable. It takes time to figure out the browser button for Speech-input user, but it is not block the submit process.
+ Alternate solutions: This issue will not break users experience, no need for alternate solutions
+ Evidence: All evidence is provided in this issue.
+ [AAT Link](https://cela-ra-aat-prod.azurewebsites.net/#/workflow/56452990-0d38-45bc-b475-a62832ce0076)

External issues tracked in: https://bugs.eclipse.org/bugs/show_bug.cgi?id=562992

**Comment by t-rufang** [comment]
@soumyatippapuram This is a WCAG2.1 bug but it's not an external bug. And we plan to fix this bug before due date 6/30/2021. The question is do we need to add this bug to AAT document?

**Comment by soumyatippapuram** [comment]
@t-rufang yes, the details of the WCAG 2.1 bug needs to be documented in AAT

**Comment by t-rufang** [comment]
@soumyatippapuram Thanks for the reply. But I still have another concern, once the conformance document is generated. It can't be modified anymore. So how do we remove the record from the conformance doc when this issue is finally fixed before due date(June 1st 2021)?

**Comment by soumyatippapuram** [comment]
@t-rufang I guess you can re-publish the Conformance Document. @umasarath52 could you please provide your insights here.

**Comment by soumyatippapuram** [comment]
@t-rufang I guess you can re-publish the Conformance Document. @umasarath52 could you please provide your insights here.

**Comment by v-umnaik** [comment]
@t-rufang 
Please note that 
All MAS rules marked as supported with exceptions need external remarks.  ​The remarks must adhere to these [guidelines](https://microsoft.sharepoint.com/teams/ceacc/SitePages/MAS%20External%20Remarks%20examples.aspx).​
All non-closed bugs and bugs with A11YExclusion must be marked as "Supported with exceptions"

**Comment by soumyatippapuram** [comment]
@t-rufang You can re-publish the Conformance Document once this bug is fixed.

**Comment by vscjavaci** [comment]
Track internal issue by AB#1749642
